OpenRA.Mods.Common/TraitsInterfaces.cs

```diff
@@ -175,8 +175,8 @@ public interface INotifyDeployComplete
 
 	public interface INotifyDeployTriggered
 	{
-		void Deploy(Actor self, bool skipMakeAnim);
-		void Undeploy(Actor self, bool skipMakeAnim);
+		void Deploy(Actor self, bool skipMakeAnim, string[] bodyNames);
+		void Undeploy(Actor self, bool skipMakeAnim, string[] bodyNames);
```

Comment:
The `WithMakeAnimation` trait and other listeners could then define the `DeployTypes` that trigger them, in addition to the `BodyNames`. The interface then doesn't need to know anything about the body names, which is an implementation detail.